resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ResourceRequestHelperSuite.scala

```diff
@@ -0,0 +1,255 @@
+    sparkConf.set(DRIVER_CORES.key, "4")
+    sparkConf.set(EXECUTOR_MEMORY.key, "4G")
+    sparkConf.set(EXECUTOR_CORES.key, "2")
+
+    ResourceRequestHelper.validateResources(sparkConf)
+  }
+
+  test("Memory defined with new config for executor") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(NEW_CONFIG_EXECUTOR_MEMORY, "30G")
+
+    val thrown = intercept[SparkException] {
+      ResourceRequestHelper.validateResources(sparkConf)
+    }
+    thrown.getMessage should include (NEW_CONFIG_EXECUTOR_MEMORY)
+  }
+
+  test("Cores defined with new config for executor") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(NEW_CONFIG_EXECUTOR_CORES, "5")
+
+    val thrown = intercept[SparkException] {
+      ResourceRequestHelper.validateResources(sparkConf)
+    }
+    thrown.getMessage should include (NEW_CONFIG_EXECUTOR_CORES)
+  }
+
+  test("Memory defined with new config, client mode") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(NEW_CONFIG_AM_MEMORY, "1G")
+
+    val thrown = intercept[SparkException] {
+      ResourceRequestHelper.validateResources(sparkConf)
+    }
+    thrown.getMessage should include (NEW_CONFIG_AM_MEMORY)
+  }
+
+  test("Memory defined with new config for driver, cluster mode") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(NEW_CONFIG_DRIVER_MEMORY, "1G")
+
+    val thrown = intercept[SparkException] {
+      ResourceRequestHelper.validateResources(sparkConf)
+    }
+    thrown.getMessage should include (NEW_CONFIG_DRIVER_MEMORY)
+  }
+
+  test("Cores defined with new config, client mode") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(DRIVER_MEMORY.key, "2G")
+    sparkConf.set(DRIVER_CORES.key, "4")
+    sparkConf.set(EXECUTOR_MEMORY.key, "4G")
+    sparkConf.set(AM_CORES.key, "2")
+    sparkConf.set(NEW_CONFIG_AM_CORES, "3")
+
+    val thrown = intercept[SparkException] {
+      ResourceRequestHelper.validateResources(sparkConf)
+    }
+    thrown.getMessage should include (NEW_CONFIG_AM_CORES)
+  }
+
+  test("Cores defined with new config for driver, cluster mode") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(NEW_CONFIG_DRIVER_CORES, "1G")
+
+    val thrown = intercept[SparkException] {
+      ResourceRequestHelper.validateResources(sparkConf)
+    }
+    thrown.getMessage should include (NEW_CONFIG_DRIVER_CORES)
+  }
+
+  test("various duplicated definitions") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(DRIVER_MEMORY.key, "2G")
+    sparkConf.set(DRIVER_CORES.key, "2")
+    sparkConf.set(EXECUTOR_MEMORY.key, "2G")
+    sparkConf.set(EXECUTOR_CORES.key, "4")
+    sparkConf.set(AM_MEMORY.key, "3G")
+
+    sparkConf.set(NEW_CONFIG_EXECUTOR_MEMORY, "3G")
+    sparkConf.set(NEW_CONFIG_AM_MEMORY, "2G")
+    sparkConf.set(NEW_CONFIG_DRIVER_MEMORY, "2G")
+
+    val thrown = intercept[SparkException] {
+      ResourceRequestHelper.validateResources(sparkConf)
+    }
+    thrown.getMessage should (
+        include(NEW_CONFIG_EXECUTOR_MEMORY) and
```

Comment:
indentation seems off here

(You seem to use both 2-space and 4-space indentation for continuation lines in different places. I think the preference is to use 2.)

Reply:
fixed